kuma/core/tests/__init__.py

```diff
@@ -23,6 +24,31 @@ def assert_shared_cache_header(response):
     assert 's-maxage' in response['Cache-Control']
 
 
+def assert_relative_uri(uri, expected_uri):
+    """
+    Assert that a URIs absolute portion matches.
```

Comment:
I think you want `relative` rather than `absolute`?

Reply:
I looked around for a few minutes to try to figure out what to call the part of the URL after the domain name. I can't find a reference, so I think we'll have to come to an agreement among ourselves.

Django has ``Model.get_absolute_url()``, which returns the path, such as ``/en-US/docs/Web/JavaScript``. We've added ``get_full_url()`` in a few places, to get the URL with host name, like ``https://developer.mozilla.org/en-US/docs/Web/JavaScript``.  I'm sad that Django uses ``get_absolute_url`` to generate a path, but I think it will be a while before I can fix that.

Python calls ``/en-US/docs/Web/JavaScript`` the path in functions like ``urlunsplit``, and calls the other bits the scheme (``https``), netloc, (``developer.mozilla.org``), query (the stuff after the question mark ``?``), and the fragment (the stuff after a hash mark ``#``).

Wikipedia has no help for what to call "the slash and everything after" on the [Uniform Resource Identifier article](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier).

All of this makes me think "call it absolute", to follow Django, but then they go an use a whole new syntax when [describing the change](https://docs.djangoproject.com/en/2.0/releases/1.9/#http-redirects-no-longer-forced-to-absolute-uris) in the release notes:

> Relative redirects are no longer converted to absolute URIs. [RFC 2616](https://tools.ietf.org/html/rfc2616.html) required the Location header in redirect responses to be an absolute URI, but it has been superseded by [RFC 7231](https://tools.ietf.org/html/rfc7231.html) which allows relative URIs in ``Location``, recognizing the actual practice of user agents, almost all of which support them.

RFC 7231 [Section 7.1.2, Location](https://tools.ietf.org/html/rfc7231.html#section-7.1.2), is where the "relative reference" terminology comes from, and refers to RFC 3986 [Section 4.2 Relative Reference](https://tools.ietf.org/html/rfc3986#section-4.2), as opposed to an absolute URI. However, "relative reference" is not a healthy search term, and mostly gets you tips and tricks in Excel.

So, here's the options, and I see them

* ``assert_relative_uri`` - supported by just about no one
* ``assert_absolute_url`` - borrow Django's relative vs Kuma's full syntax
* ``assert_relative_reference`` - borrow from RFC 3986

My weak preference is the last one, with a link to RFC 3986.

Reply:
Wow, that's a nice summary, thanks! I have a weak preference for the last one as well, but I can accept anything that you decide.

Reply:
OK, going with ``assert_relative_reference``. That will help me quickly get rid of it when we're on Django 1.11!